test/integration/physics_system.cc

```diff
@@ -1332,4 +1332,19 @@ TEST_F(PhysicsSystemFixture, NestedModelIndividualCanonicalLinks)
   modelIt = postUpModelPoses.find("model_01");
   EXPECT_NE(postUpModelPoses.end(), modelIt);
   EXPECT_NEAR(zExpected, modelIt->second.Z(), 1e-2);
+
+  // model_10, model_11 and model_12 share the same canonical link (link_11),
+  // which is a floating link. So, link_11 should fall due to gravity, which
+  // also means that the frames for model_10, model_11 and model_12 should fall
+  auto model10It = postUpModelPoses.find("model_10");
+  EXPECT_NE(postUpModelPoses.end(), model10It);
+  EXPECT_NEAR(zExpected, model10It->second.Z(), 1e-2);
+  auto model11It = postUpModelPoses.find("model_11");
+  EXPECT_NE(postUpModelPoses.end(), model11It);
+  auto model12It = postUpModelPoses.find("model_12");
+  EXPECT_NE(postUpModelPoses.end(), model12It);
+  // (since model_11 and model_12 are nested, their poses are computed w.r.t.
+  // model_10)
+  EXPECT_DOUBLE_EQ(0.0, model11It->second.Z());
+  EXPECT_DOUBLE_EQ(0.0, model12It->second.Z());
```

Comment:
We could use `gazebo::worldPose` from `Util.hh` to check that they're also at `zExpected` if we want.

Reply:
https://github.com/ignitionrobotics/ign-gazebo/pull/736/commits/11a97df6417c8fde301d63817c81ca816e6df58e